libs/src/api/file.js

```diff
@@ -68,12 +68,21 @@ class File extends Base {
 
     const gateways = creatorNodeGateways.concat(publicGateways)
 
-    return retry(async () => {
+    return retry(async (bail) => {
       try {
-        const { response } = await raceRequests(urls, callback, {
+        const { response, errored } = await raceRequests(urls, callback, {
           method: 'get',
           responseType
         }, /* timeout */ null)
+
+        const allUnauthorized =
+          response === undefined &&
+          errored.every(error => error.response.status === 403)
+        if (allUnauthorized) {
+          // In the case for a 403, do not retry fetching
+          bail(new Error('Unauthorized'))
```

Comment:
instead of returning `new Error`, what about returning `error` from `errored` or reconstructing the error to get the response status code and other metadata?

Reply:
nvm, i see you got this directly from the async-retry docs haha

Reply:
yeah, i mean i don't love it, but i think passing through the entire response is going to be somewhat funky. this is definitely a stop gap.

Reply:
and separately with the "sdk" epic, i think this logic could be rewritten from the ground up. it's bifurcated in AudiusBackend and here too, which is less than ideal

Reply:
yeah am seeing that, all good. resolved this discussion